docs/articles/nunit/release-notes/Nunit4.0-MigrationGuide.md

```diff
@@ -0,0 +1,169 @@
+---
+uid: nunit40migrationguide
+---
+
+# NUnit 4.0 Migration Guide
+
+NUnit 4.0 has a few [breaking changes](../release-notes/breaking-changes.md#nunit-40)
+making it neither binary nor source code compatible with NUnit 3.14.0
+
+* Change to [Classic Asserts](../writing-tests/assertions/assertion-models/classic.md)
+* Removal of `Assert.That` overloads with _format_ specification and `params`.
+
+## Classic Assert migration
+
+There are different ways to migrate these to NUnit 4.0
+
+* Convert Classic Assert to the [Constraint model](../writing-tests/assertions/assertion-models/constraint.md)
+* Update source code to new namespace and class name
+* Using `global using` aliases
+
+In the sections below we use the following simple test as an example:
+
+```csharp
+public class Tests
+{
+    [Test]
+    public void TestSomeCalculation()
+    {
+        int actual = SomeCalculation();
+        Assert.AreEqual(42, actual, "Expected {0} to be 42", actual);
+        StringAssert.StartsWith("42", actualText, "Expected '{0}' to start with '42'", actualText);
+    }
+
+    private static int SomeCalculation() => 42;
+}
+```
+
+### Convert Classic Assert to Constraint Model
+
+Although the code can be converted manually, that is a lot of work.
+
+Luckily, the [NUnit.Analyzer](https://www.nuget.org/packages/NUnit.Analyzers) has had rules and associated code fixes for a while now.
+Version _3.10.0_ knows about the 2nd non-backward compatible change and will convert the _format_ specification and `params` into a `FormattableString`.
```

Comment:
@mikkelbu You might want to release that version before this is merged/published.

Reply:
Good catch; I can hold off on this until I hear from @mikkelbu.

Reply:
I decided the other contents were valuable enough and that this could be a living guide, so I didn't want to rush @mikkelbu before merging. If we get any questions on it, it can be answered easily eonugh.

Reply:
I'll see if I can get an release out tomorrow (work is taking most of my time/energy at the moment)